tests/utils/test_dependency_constraints.py

```
@@ -0,0 +1,162 @@
+    # `!=0.1.13` excludes only 0.1.13 itself under PEP 440, so post-releases still
+    # resolve; the ABI-divergent ones are excluded by name.
+    for bad in ("0.1.13", "0.1.13.post0", "0.1.13.post1"):
+        assert not spec.contains(bad)
+
+
+def test_no_unlisted_tvm_ffi_references():
+    """A new site naming tvm-ffi must be classified, not silently added."""
+    try:
+        grep = subprocess.run(
+            # Both the PyPI name and the VCS URL form used by overrides.
+            ["git", "grep", "-lE", "apache[-/]tvm-ffi", "--", ".", ":!3rdparty"],
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)